### PR TITLE
Add API endpoints for workflows [SCI-4831]

### PR DESCRIPTION
**Pull request by aignatov-bio**
Jira ticket: [SCI-4831](https://biosistemika.atlassian.net/browse/SCI-4831)

### What was done
Add API endpoints for workflows

### CODE REVIEW

File: app/serializers/api/v1/task_serializer.rb [hunk 1]
```diff
@@ -4,14 +4,39 @@ module Api
...
 
+      def status_id
+        object.my_module_status_id
+      end
+
+      def status_name
+        object.my_module_status.name
+      end
+
+      def prev_status_id
```
Comment (urbanrotnik) [inline app/serializers/api/v1/task_serializer.rb:24]: You can use belongs_to for previous and next status with its serializer...?

**Review by urbanrotnik — CHANGES_REQUESTED** [review]
Check comments and implement request specs for new controllers. You can add also tests for schema matching

File: app/serializers/api/v1/task_serializer.rb (hunk 1 above)
Comment (aignatov-bio) [reply app/serializers/api/v1/task_serializer.rb:28]: It does not belong to my_module

File: app/controllers/api/v1/tasks_controller.rb [hunk 2]
```diff
@@ -59,7 +60,7 @@ def activities
       def task_params
         raise TypeError unless params.require(:data).require(:type) == 'tasks'
 
-        params.require(:data).require(:attributes).permit(%i(name x y description state))
```
Comment (urbanrotnik) [inline app/controllers/api/v1/tasks_controller.rb:62]: Can you set not first my_module_status on the `create` method also?  Or will fail with validation for check_status?

**Review by urbanrotnik — APPROVED** [review]
Check one comment, if it is covered, merge

File: app/serializers/api/v1/task_serializer.rb (hunk 1 above)
Comment (urbanrotnik) [reply app/serializers/api/v1/task_serializer.rb:28]: But status `belongs_to`...

**Pull request merged**